src/lib/utils/meta_tables/meta_system_utilization_table.cpp

```diff
@@ -0,0 +1,337 @@
+  const auto process_memory_usage = _get_process_memory_usage();
+
+  output_table->append({system_cpu_ticks, process_cpu_ticks, total_ticks, load_avg.load_1_min, load_avg.load_5_min,
+                        load_avg.load_15_min, system_memory_usage.free_memory, system_memory_usage.available_memory,
+                        process_memory_usage.virtual_memory, process_memory_usage.physical_memory});
+
+  return output_table;
+}
+
+/*
+  * Returns the load average values for 1min, 5min, and 15min.
+*/
+MetaSystemUtilizationTable::LoadAvg MetaSystemUtilizationTable::_get_load_avg() {
+#ifdef __linux__
+  std::ifstream load_avg_file;
+  load_avg_file.open("/proc/loadavg", std::ifstream::in);
+
+  if (!load_avg_file.is_open()) {
+    Fail("Unable to open /proc/loadavg");
+  }
+
+  std::string load_avg_value;
+  std::vector<float> load_avg_values;
+  for (int value_index = 0; value_index < 3; ++value_index) {
+    std::getline(load_avg_file, load_avg_value, ' ');
+    load_avg_values.push_back(std::stof(load_avg_value));
+  }
+  load_avg_file.close();
+
+  return {load_avg_values[0], load_avg_values[1], load_avg_values[2]};
+#endif
+
+#ifdef __APPLE__
+  loadavg load_avg;
+  size_t size = sizeof(load_avg);
+  if (sysctlbyname("vm.loadavg", &load_avg, &size, nullptr, 0) != 0) {
+    Fail("Unable to call sysctl vm.loadavg");
+  }
+
+  return {static_cast<float>(load_avg.ldavg[0]) / static_cast<float>(load_avg.fscale),
+          static_cast<float>(load_avg.ldavg[1]) / static_cast<float>(load_avg.fscale),
+          static_cast<float>(load_avg.ldavg[2]) / static_cast<float>(load_avg.fscale)};
+#endif
+
+  Fail("Method not implemented for this platform");
+}
+
+/*
+  * Returns the time in ns since an arbitrary point in the past.
+*/
+int64_t MetaSystemUtilizationTable::_get_total_time() {
+/* CLOCK_MONOTONIC_RAW:
+ * Similar to CLOCK_MONOTONIC, but provides access to a raw hard‐
+ * ware-based time that is not subject to NTP adjustments or the
+ * incremental adjustments performed by adjtime(3).  This clock
+ * does not count time that the system is suspended.
+*/
+#ifdef __linux__
+  struct timespec time_spec;
+  const auto ret = clock_gettime(CLOCK_MONOTONIC_RAW, &time_spec);
+  if (ret == -1) {
+    Fail("An error occured while fetching the time");
+  }
+
+  const auto total_ns = time_spec.tv_sec * std::nano::den + time_spec.tv_nsec;
+
+  return total_ns;
+#endif
+
+#ifdef __APPLE__
+  const auto total_ns = clock_gettime_nsec_np(CLOCK_MONOTONIC_RAW);
+  if (total_ns == 0) {
+    Fail("An error occured while fetching the time");
+  }
+  return total_ns;
+#endif
+
+  Fail("Method not implemented for this platform");
+}
+
+/*
+ * Returns the time in ns that ALL processes have spend on the CPU
+ * since an arbitrary point in the past.
+*/
+int64_t MetaSystemUtilizationTable::_get_system_cpu_time() {
+// The amount of time is measured in units of clock ticks.
+// sysconf(_SC_CLK_TCK) can be used to obtain the right value.

```

Comment:
What is the purpose of this comment? Above you say the time is in `ns`. Do you mean the resolution here?